src/vs/workbench/contrib/terminal/electron-browser/terminalInstanceService.ts

```diff
@@ -157,10 +157,10 @@ export class TerminalInstanceService extends Disposable implements ITerminalInst
 		return getMainProcessParentEnv();
 	}
 
-	public setTerminalLayoutInfo(layoutInfo: ITerminalsLayoutInfoById): void {
+	public setTerminalLayoutInfo(layoutInfo?: ITerminalsLayoutInfoById): void {
 		const args: ISetTerminalLayoutInfoArgs = {
 			workspaceId: this._getWorkspaceId(),
-			tabs: layoutInfo.tabs
+			tabs: layoutInfo ? layoutInfo.tabs : []
```

Comment:
Ideally this would also pass through `undefined` and then on ptyService we would free the memory:

```ts
	async setTerminalLayoutInfo(args: ISetTerminalLayoutInfoArgs): Promise<void> {
		if (args.tabs) {
			this._workspaceLayoutInfos.delete(args.workspaceId);
		} else {
			this._workspaceLayoutInfos.set(args.workspaceId, args);
		}
	}
```

Reply:
I think we want to do the opposite of the above snippet, but i'm sure that's what you meant

Reply:
oh yeah 🤦

Reply:
leaving as is bc as we discussed, would be a lot of changes and they could impact remote